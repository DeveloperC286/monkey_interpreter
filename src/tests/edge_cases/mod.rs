use super::*;

#[template]
#[rstest(code, snapshot_name, case("", "edge_case_1"))]
fn edge_cases(code: &str, snapshot_name: &str) {}

#[apply(edge_cases)]
fn test_edge_cases_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(edge_cases)]
fn test_edge_cases_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(edge_cases)]
fn test_edge_cases_evaluation(code: &str, snapshot_name: &str) {
    assert_evaluation!(code, snapshot_name);
}

#[apply(edge_cases)]
fn test_edge_cases_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
