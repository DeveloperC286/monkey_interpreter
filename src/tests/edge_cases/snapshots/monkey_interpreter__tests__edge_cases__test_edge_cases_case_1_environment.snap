---
source: src/tests/edge_cases/mod.rs
expression: evaluator
---
Evaluator {
    environment: Environment {
        variables: {},
        sub_environment: None,
    },
}
